plugins/outputs/groundwork/log_adapter.go

```diff
@@ -0,0 +1,84 @@
+package groundwork
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+)
+
+// NewLogger creates telegraf.Logger adapter for slog.Logger
+func NewLogger(l telegraf.Logger) *slog.Logger {
+	return slog.New(&TlgHandler{Log: l})
+}
+
+// TlgHandler translates slog.Record into telegraf.Logger call
+// inspired by https://github.com/golang/example/blob/master/slog-handler-guide/README.md
+type TlgHandler struct {
```

Comment:
Please don't export these things! Maybe even kill off the `NewLogger` function completely as it only wraps a single line. However, I see the value in keeping `log/slog` only in here...

Reply:
yes, there is the value to keep `log/slog` only in here
and done TlgHandler => tlgHandler